File: Smartystreets_iOS_SDK/Examples/Objective_C_Examples/Sources/SSZipCodeSingleLookupExample.m
#import "SSZipCodeSingleLookupExample.h"

@implementation SSZipCodeSingleLookupExample

- (NSString*)run {
    id<SSCredentials> mobile = [[SSSharedCredentials alloc] initWithId:kSSSmartyWebsiteKey hostname:kSSHost];
    SSZipCodeClient *client = [[SSZipCodeClientBuilder alloc] initWithSigner:mobile].build;
//    SSZipCodeClient *client = [[SSZipCodeClientBuilder alloc] initWithAuthId:kSSAuthId authToken:kSSAuthToken].build;
    
    SSZipCodeLookup *lookup = [[SSZipCodeLookup alloc] init];
    lookup.city = @"Mountain View";
    lookup.state = @"California";
    
    NSError *error = nil;
    [client sendLookup:lookup error:&error];
    
    if (error != nil) {
        NSLog(@"Domain: %@", error.domain);
        NSLog(@"Error Code: %i", (int)error.code);
        NSLog(@"Description: %@", [error localizedDescription]);
        return @"Error sending request";
    }
    
    SSResult *result = lookup.result;
    NSArray<SSZipCode*> *zipCodes = result.zipCodes;
    NSArray<SSCity*> *cities = result.cities;
    
    NSMutableString *output = [[NSMutableString alloc] init];
    
    if (cities == nil && zipCodes == nil) {
        [output appendString:@"Error getting cities and zip codes."];
        return output;
    }
    
    for (SSCity *city in cities) {
        [output appendString:[@"\nCity: " stringByAppendingString:city.city]];
        [output appendString:[@"\nState: " stringByAppendingString:city.state]];
        [output appendString:[@"\nMailable City: " stringByAppendingString:city.mailableCity ? @"YES" : @"NO"]];
        [output appendString:@"\n"];
    }
    
    for (SSZipCode *zip in zipCodes) {
        [output appendString:[@"\nZIP Code: " stringByAppendingString:zip.zipCode]];
        NSNumber *latitude = [NSNumber numberWithDouble:zip.latitude];
        NSNumber *longitute = [NSNumber numberWithDouble:zip.longitude];
        [output appendString:[@"\nLatitude: " stringByAppendingString:[latitude stringValue]]];
        [output appendString:[@"\nLongitude: " stringByAppendingString:[longitute stringValue]]];
        [output appendString:@"\n"];
    }
    
    return output;
}

@end

File: Examples_Swift/Examples_Swift/SSZipCodeSingleLookupExample.swift
import Foundation
import Smartystreets_iOS_SDK

class SSZipCodeSingleLookupExample {

    func run() -> String {
//        let mobile = SSSharedCredentials(id: "key", hostname: "host")
//        let client = SSZipCodeClientBuilder(signer: mobile).build()
        let client = SSZipCodeClientBuilder(authId: "AUTH_ID", //TODO: figure out why can't send request in Swift
                                            authToken: "AUTH_TOKEN").build()
        
        let lookup = SSZipCodeLookup()
        lookup.city = "Mountain View"
        lookup.state = "California"
        
        var error: NSError?
        client!.send(lookup, error: &error)
        
        if (error != nil) {
            return "Error sending request"
        }
        
        let result: SSResult = lookup.result
        let zipCodes = result.zipCodes
        let cities = result.cities
        
        var output: String = String()
        
        if (cities == nil && zipCodes == nil) {
            output += "Error getting cities and zip codes."
            return output
        }
        
        for city in cities! {
            output += "\nCity: " + (city as! SSCity).city
            output += "\nState: " + (city as! SSCity).state
            output += "\nMailable City: " + ((city as! SSCity).mailableCity ? "YES" : "NO")
            output += "\n"
        }
        
        for zip in zipCodes! {
            output += "\nZIP Code: " + (zip as! SSZipCode).zipCode
            output += "\nLatitude: " + String(format:"%@", (zip as! SSZipCode).latitude)
            output += "\nLongitude: " + String(format:"%@", (zip as! SSZipCode).longitude)
            output += "\n"
        }
        
        return output
    }

}